go/vm/lfvm/memory.go

```diff
@@ -44,14 +44,33 @@ func toValidMemorySize(size uint64) uint64 {
 	return fullWordsSize
 }
 
+const (
+	// Maximum memory size allowed
+	// This magic number comes from 'core/vm/gas_table.go' 'memoryGasCost' in geth
```

Comment:
You could de-magicfy it a bit by stating that this is the largest amount of memory that can be allocated such that the expansion costs do not exceed `MaxInt64`.